src/FuncSharp.Examples/IOption.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace FuncSharp.Examples
+{
+    public class IOptionUsages
+    {
+        public IOption<decimal> Divide(decimal number, decimal divisor)
+        {
+            return divisor.ToOption().Where(d => d != 0).Map(d => number / d);
+        }
+
+        private void TransformingOptionValuesWithMap(decimal number, decimal divisor)
+        {
+            IOption<decimal> divisionResult = Divide(number, divisor);
+            IOption<decimal> roundedDivisionResult = divisionResult.Map(r => Math.Round(r));
+            IOption<string> stringifiedDivisionResult = divisionResult.Map(r => r.ToString());
+        }
+
+        private void HandlingNestedOptionsWithFlatMap(decimal number, decimal divisor)
```

Comment:
I'd use two divisors here, because with single divisor, you could implement this better (divisor is zero or not, so single option would suffice theoretically).